homeassistant/components/air_quality/foobot.py

```diff
@@ -0,0 +1,220 @@
+"""
+Support for the Foobot indoor air quality monitor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/air_quality.foobot/
+"""
+import asyncio
+import logging
+from datetime import datetime, timedelta
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.components.air_quality import (
+    AirQualityEntity, PLATFORM_SCHEMA, PROP_TO_ATTR)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.const import (
+    ATTR_TIME, ATTR_TEMPERATURE, CONF_TOKEN, CONF_USERNAME)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_FOOBOT_INDEX = 'foobot_index'
+ATTR_HUMIDITY = 'humidity'
+ATTR_VOC = 'volatile_organic_compound'
+
+PARALLEL_UPDATES = 1
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+})
+
+PROP_TO_ATTR.update({'foobot_index': ATTR_FOOBOT_INDEX,
```

Comment:
We're not allowed to modify this dict. It's considered constant and it's used by the base class in the state attributes property. We would affect all platforms by doing this. Please define a new dictionary here.